frontend/pages/layouts/index.js

```diff
@@ -1,12 +1,38 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import { Link } from 'react-router-dom';
 import { loadLayouts } from '~/store/actions/layouts';
 import { getLayouts } from '~/store/getters/layouts';
 import LayoutCard from '~/components/layouts/LayoutCard';
+import InfiniteScroll from "react-infinite-scroller";
+
+
+const pageSize = 10
 
 export default () => {
   const layouts = useSelector(getLayouts())
+  const dispatch = useDispatch()
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const [hasMore, setHasMore] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
+  const isLastPage = currentPage*pageSize > layouts.length
+
+
+  function loadMoreLayouts(){
+    if (!hasMore || isLoading) {
+      return;
+    }
+    dispatch(loadLayouts((currentPage-1) * pageSize, pageSize))
+    setIsLoading(true);
+    setHasMore(false)
+    if (!isLastPage) {
+      // if timeout is not used, scroll don't move up after loading data and send continuous calls to fetch data (infinite calls)
+      setTimeout(function () { setHasMore(true); }, 200);
```

Comment:
No need to use set timeout, dispatch returns a promise, before calling the promise `setLoading(true)` and when the promise resolves or rejects `setLoading(false)`